### PR TITLE
Rechunk script

### PR DESCRIPTION
**Pull request by joshmoore**
Permits copying an array from one Zarr store to another while applying a new chunk size. Dask `from_zarr` and `to_zarr` are used to perform this in parallel. For long running tasks, a dashboard can be activated with `--distributed` to watch the processing occur. Multiscale metadata will need to be updated for the Zarr group _after_ the fact.

### CODE REVIEW

File: src/scripts/rechunk.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--dimensions", type=int, default="5", metavar="DIMS",
+                        help="number of chunks to parse (%(default)s)")
+    parser.add_argument("--distributed", action="store_true",
+                        help="enable distributed dashboard (%(default)s)")
+    parser.add_argument("source_array",
+                        help="array to copy from. must exist.")
+    parser.add_argument("target_array",
+                        help="array to write to. may not exist.")
+    parser.add_argument("chunks", default="1024,1024,1,1,1",
```
Comment (mtbc) [inline src/scripts/rechunk.py:48]: I expected to find the non-1's at the start, not the end.

Comment (joshmoore) [reply src/scripts/rechunk.py]: Ta.

**Comment by mtbc** [comment]
```
(zarr) mtbc@ls28101:~/src/omero-ms-zarr/zarr$ ../src/scripts/rechunk.py -h
  File "../src/scripts/rechunk.py", line 49
    help=("comma-separated string of chunk sizes "
    ^
IndentationError: unexpected indent
(zarr) mtbc@ls28101:~/src/omero-ms-zarr/zarr$ /usr/bin/env python --version
Python 3.7.7
```

File: src/scripts/rechunk.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--dimensions", type=int, default="5", metavar="DIMS",
+                        help="number of chunks to parse (%(default)s)")
+    parser.add_argument("--distributed", action="store_true",
+                        help="enable distributed dashboard (%(default)s)")
+    parser.add_argument("source_array",
+                        help="array to copy from. must exist.")
+    parser.add_argument("target_array",
+                        help="array to write to. may not exist.")
```
Comment (mtbc) [inline src/scripts/rechunk.py:47]: Seems it better not or I get,
> ValueError: path '' contains an array

File: src/scripts/rechunk.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+def timeit(method):
+    def timed(*args, **kw):
+        ts = time.time()
+        result = method(*args, **kw)
+        te = time.time()
+        if 'log_time' in kw:
+            name = kw.get('log_name', method.__name__.upper())
+            kw['log_time'][name] = int((te - ts) * 1000)
+        else:
+            print('%r  %2.2f ms' %
```
Comment (mtbc) [inline src/scripts/rechunk.py:17]:
```suggestion
            print('%s  %2.2f ms' %
```

File: src/scripts/rechunk.py (hunk 2 above)
Comment (joshmoore) [reply src/scripts/rechunk.py:47]: Is this if the target already already exists?

File: src/scripts/rechunk.py (hunk 3 above)
Comment (joshmoore) [reply src/scripts/rechunk.py:17]: `%r` is for using `repr()`. I'm not exactly sure why the snippet author chose that, but it certainly doesn't hurt.

File: src/scripts/rechunk.py (hunk 2 above)
Comment (mtbc) [reply src/scripts/rechunk.py:47]: It is, yes.

File: src/scripts/rechunk.py (hunk 3 above)
Comment (mtbc) [reply src/scripts/rechunk.py:17]: Basically I wondered why we want to see quotes around the printed words.

File: src/scripts/rechunk.py (hunk 2 above)
Comment (joshmoore) [reply src/scripts/rechunk.py:47]: So the exception is bad, but matches the help text, right?

Comment (mtbc) [reply src/scripts/rechunk.py:47]: No, "must not exist" would match the exception?

Comment (joshmoore) [reply src/scripts/rechunk.py:47]: That's not how I parse it, but ok. I'll update.

Comment (mtbc) [reply src/scripts/rechunk.py]: Oh, haha, I thought it was saying I didn't have to pre-create it! Thank you.

**Pull request merged**